### PR TITLE
[FEAT](backend) - read product API

### PR DESCRIPTION
**Pull request by markjoshuacarmen**
## Ticket/s
[[API] Read Product](https://trello.com/c/0fmhz3gA/33-api-read-product)

## Changes
- fix missing import for product schema
- add get product controller

## Screenshot

image

## How to test?
   1. Create product 
   2. Use this link
   ```
curl --location 'http://localhost:3001/products/create' \
--header 'Authorization: Bearer' \
--header 'Content-Type: application/json' \
--data '{
    "productID": "product1",
    "productImg": {
        "ImgURL": "https://example.com/image.jpg",
        "ImgAttch": "base64encodedimage"
    },
    "productName": "Product Name",
    "productInfo": "Product Information",
    "productPrice": 199.99,
    "productInventory": 32,
    "productCategory": [
        "sample",
        "sample1",
        "sample2"
    ]
}'
```
   4. Copy the product Id in Db
   5. Copy the product token in login then paste in the Authorization Bearer in Get request
   6. Import cURL for the GET request.
```
curl --location --request GET 'http://localhost:3001/products/64c9c680e8adccd29430e880' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InNhbXBsZSIsInN1YiI6IjY0YzljMThlOGYwODY3NmU3YzgyYzA3NCIsImlhdCI6MTY5MDk0Mzk3OSwiZXhwIjoxNjkxMDMwMzc5fQ.HV-9LG--HZ-73qwY-xX0rkM3wP8ky5Hzu8FQOQ48YzQ' \
--header 'Content-Type: application/json' \
--data '{
    "productID": "asdasd",
    "productImg": {
        "ImgURL": "https://example.com/image.jpg",
        "ImgAttch": "base64encodedimage"
    },
    "productName": "test",
    "productInfo": "Product Information",
    "productPrice": 23,
    "productInventory": 32,
    "productCategory": [
        "sample",
        "sample1",
        "sample2"
    ]
}
'
```
   7. replace <place id here> placeholder with product _id from mongoDB
curl --location 'localhost:3001/products/<place id here>'

### CODE REVIEW

File: apps/api/src/products/products.service.ts [hunk 1]
```diff
@@ -11,7 +15,25 @@ export class ProductsService {
...
   }
+
+  async findOne(id: string): Promise<Product> {
+    const product = await this.productModel.findOne({ _id: id });
+
+    if (!product) {
+      throw new NotFoundException('Product not found');
+    }
+
+    //console.log(merchant);
```
Comment (sandrex-SGV) [inline apps/api/src/products/products.service.ts]:
```suggestion

```

File: apps/api/src/products/products.service.ts [hunk 2]
```diff
@@ -11,7 +15,25 @@ export class ProductsService {
   ) {}
 
   async create(createProductDto: CreateProductDto): Promise<Product> {
+    const productAlreadyExists = await this.productModel
+      .findOne({ productID: createProductDto.productID })
+      .exec();
+    if (productAlreadyExists) {
+      throw new BadRequestException('Product ID already exists');
```
Comment (sandrex-SGV) [inline apps/api/src/products/products.service.ts:25]:
```suggestion
      throw new BadRequestException('Product already exist');
```

**Review by sandrex-SGV — CHANGES_REQUESTED** [review]
Please check requested changes. many thx

**Review by TreshAnn — APPROVED** [review]
LGTM!

**Review by sandrex-SGV — APPROVED** [review]

**Pull request merged**